mqcloudv1/mqcloud_v1_integration_test.go

```diff
@@ -725,13 +727,127 @@ var _ = Describe(`MqcloudV1 Integration Tests`, func() {
+		})
+		It(`ListVirtualPrivateEndpointGateways(listVirtualPrivateEndpointGatewaysOptions *ListVirtualPrivateEndpointGatewaysOptions) with pagination`, func() {
+			listVirtualPrivateEndpointGatewaysOptions := &mqcloudv1.ListVirtualPrivateEndpointGatewaysOptions{
+				ServiceInstanceGuid: core.StringPtr(config["SERVICE_INSTANCE_CAPACITY_GUID"]),
+				TrustedProfile:      core.StringPtr(config["TRUSTED_PROFILE"]),
+				Start:               core.StringPtr(""),
+				Limit:               core.Int64Ptr(int64(10)),
+			}
+
+			listVirtualPrivateEndpointGatewaysOptions.TrustedProfile = nil
+			listVirtualPrivateEndpointGatewaysOptions.Start = nil
+			listVirtualPrivateEndpointGatewaysOptions.Limit = core.Int64Ptr(1)
+
+			var allResults []mqcloudv1.VirtualPrivateEndpointGatewayDetails
+			for {
+
+				virtualPrivateEndpointGatewayDetailsCollection, response, err := mqcloudService.ListVirtualPrivateEndpointGateways(listVirtualPrivateEndpointGatewaysOptions)
+
+				Expect(err).To(BeNil())
+				Expect(response.StatusCode).To(Equal(200))
+				Expect(virtualPrivateEndpointGatewayDetailsCollection).ToNot(BeNil())
+				allResults = append(allResults, virtualPrivateEndpointGatewayDetailsCollection.VirtualPrivateEndpointGateways...)
+
+				listVirtualPrivateEndpointGatewaysOptions.Start, err = virtualPrivateEndpointGatewayDetailsCollection.GetNextStart()
+				Expect(err).To(BeNil())
+				if listVirtualPrivateEndpointGatewaysOptions.Start == nil {
+
+					break
+				}
+				fmt.Println("START=" + *listVirtualPrivateEndpointGatewaysOptions.Start)
+			}
+			fmt.Fprintf(GinkgoWriter, "Retrieved a total of %d item(s) with pagination.\n", len(allResults))
+		})
+		It(`ListVirtualPrivateEndpointGateways(listVirtualPrivateEndpointGatewaysOptions *ListVirtualPrivateEndpointGatewaysOptions) using VirtualPrivateEndpointGatewaysPager`, func() {
+			listVirtualPrivateEndpointGatewaysOptions := &mqcloudv1.ListVirtualPrivateEndpointGatewaysOptions{
+				ServiceInstanceGuid: core.StringPtr(config["SERVICE_INSTANCE_CAPACITY_GUID"]),
+				TrustedProfile:      core.StringPtr(config["TRUSTED_PROFILE"]),
+				Limit:               core.Int64Ptr(int64(10)),
+			}
+
+			// Test GetNext().
+			listVirtualPrivateEndpointGatewaysOptions.TrustedProfile = nil
+			pager, err := mqcloudService.NewVirtualPrivateEndpointGatewaysPager(listVirtualPrivateEndpointGatewaysOptions)
+			Expect(err).To(BeNil())
+			Expect(pager).ToNot(BeNil())
+
+			var allResults []mqcloudv1.VirtualPrivateEndpointGatewayDetails
+			for pager.HasNext() {
+				nextPage, err := pager.GetNext()
+				Expect(err).To(BeNil())
+				Expect(nextPage).ToNot(BeNil())
+				allResults = append(allResults, nextPage...)
+			}
+
+			// Test GetAll().
+			pager, err = mqcloudService.NewVirtualPrivateEndpointGatewaysPager(listVirtualPrivateEndpointGatewaysOptions)
+			Expect(err).To(BeNil())
+			Expect(pager).ToNot(BeNil())
+
+			allItems, err := pager.GetAll()
+			Expect(err).To(BeNil())
+			//		Expect(allItems).ToNot(BeNil())
```

Comment:
Is there any potential issue with checking the vpeg list items here? Just wanted to ensure we're not missing anything.
```suggestion
			Expect(allItems).ToNot(BeNil())
```

Reply:
I am not sure but can check and get back @stalleyj

Reply:
Oh i see, But by default we are using pagers in the terraform provider for the vpeg data source read operation.

See the snippet
```
	var pager *mqcloudv1.VirtualPrivateEndpointGatewaysPager
	pager, err = mqcloudClient.NewVirtualPrivateEndpointGatewaysPager(listVirtualPrivateEndpointGatewaysOptions)
```

So will be worth checking the sdk/provider functions are working well.

Reply:
Agreed, it isn't ideal so will do further further testing

Reply:
You can uncomment this line.

As its passing now. (just checked)


```
go test -tags=integration                                                                                             ✔ 
Running Suite: MqcloudV1 Suite
==============================
Random Seed: 1732299431
Will run 240 of 240 specs

••
------------------------------
• [SLOW TEST:14.099 seconds]
MqcloudV1 Integration Tests
/Users/shuaib/go/src/github.com/IBM/mqcloud-go-sdk/mqcloudv1/mqcloud_v1_integration_test.go:41
  CreateVirtualPrivateEndpointGateway - Create a new virtual private endpoint gateway
  /Users/shuaib/go/src/github.com/IBM/mqcloud-go-sdk/mqcloudv1/mqcloud_v1_integration_test.go:730
    CreateVirtualPrivateEndpointGateway(createVirtualPrivateEndpointGatewayOptions *CreateVirtualPrivateEndpointGatewayOptions)
    /Users/shuaib/go/src/github.com/IBM/mqcloud-go-sdk/mqcloudv1/mqcloud_v1_integration_test.go:734
------------------------------
START=eyJhY2NvdW50X2lkIjoiNWQ1ZmYyYTkwMDFjNDA1NWFiMTQwOGU5YmYxODVmNDgiLCJ1c2VyX2lkIjoiaWFtLVNlcnZpY2VJZC05MGI2MWYwMy1jNTMyLTQxNmMtYjQ4MS0yNzUxNzVlNTY1NDciLCJsaW1pdCI6MSwidGltZW91dCI6MCwic29ydCI6ImNybjphc2MiLCJpc19kZWxldGVkIjoiZmFsc2UiLCJpc19yZWNsYWltZWQiOiJmYWxzZSIsImlzX3B1YmxpYyI6ImZhbHNlIiwiY2FuX3RhZyI6ImZhbHNlIiwicXVlcnkiOiJ0YWdzOjFlZWY4ZjBmLWFjYTctNGEzZS05MDc0LTM5MGZmMjI1N2UzYyBBTkQgdHlwZTplbmRwb2ludC1nYXRld2F5IiwiZmllbGRzIjpbImNybiIsIm5hbWUiLCJmYW1pbHkiLCJ0eXBlIiwiYWNjb3VudF9pZCJdLCJpc19wcm9qZWN0X3Jlc291cmNlIjoiZmFsc2UiLCJjcm9zc0FjY291bnRzIjpmYWxzZSwic2VhcmNoQWZ0ZXIiOlsiY3JuOnYxOmJsdWVtaXg6cHVibGljOmlzOmV1LWRlOmEvNWQ1ZmYyYTkwMDFjNDA1NWFiMTQwOGU5YmYxODVmNDg6OmVuZHBvaW50LWdhdGV3YXk6cjAxMC03NGFmYWE0Ni00ZGJkLTQ0YjAtODFhNi04ZjM2NDY0NzE0ZDgiXX0=
• [SLOW TEST:9.314 seconds]
MqcloudV1 Integration Tests
/Users/shuaib/go/src/github.com/IBM/mqcloud-go-sdk/mqcloudv1/mqcloud_v1_integration_test.go:41
  ListVirtualPrivateEndpointGateways - Get a list of information for all virtual private endpoint gateways
  /Users/shuaib/go/src/github.com/IBM/mqcloud-go-sdk/mqcloudv1/mqcloud_v1_integration_test.go:754
    ListVirtualPrivateEndpointGateways(listVirtualPrivateEndpointGatewaysOptions *ListVirtualPrivateEndpointGatewaysOptions) with pagination
    /Users/shuaib/go/src/github.com/IBM/mqcloud-go-sdk/mqcloudv1/mqcloud_v1_integration_test.go:758
------------------------------
• [SLOW TEST:11.248 seconds]
MqcloudV1 Integration Tests
/Users/shuaib/go/src/github.com/IBM/mqcloud-go-sdk/mqcloudv1/mqcloud_v1_integration_test.go:41
  ListVirtualPrivateEndpointGateways - Get a list of information for all virtual private endpoint gateways
  /Users/shuaib/go/src/github.com/IBM/mqcloud-go-sdk/mqcloudv1/mqcloud_v1_integration_test.go:754
    ListVirtualPrivateEndpointGateways(listVirtualPrivateEndpointGatewaysOptions *ListVirtualPrivateEndpointGatewaysOptions) using VirtualPrivateEndpointGatewaysPager
    /Users/shuaib/go/src/github.com/IBM/mqcloud-go-sdk/mqcloudv1/mqcloud_v1_integration_test.go:790
------------------------------
•
------------------------------
• [SLOW TEST:5.616 seconds]
MqcloudV1 Integration Tests
/Users/shuaib/go/src/github.com/IBM/mqcloud-go-sdk/mqcloudv1/mqcloud_v1_integration_test.go:41
  DeleteVirtualPrivateEndpointGateway - Delete a specific virtual private endpoint gateway
  /Users/shuaib/go/src/github.com/IBM/mqcloud-go-sdk/mqcloudv1/mqcloud_v1_integration_test.go:927
    DeleteVirtualPrivateEndpointGateway(deleteVirtualPrivateEndpointGatewayOptions *DeleteVirtualPrivateEndpointGatewayOptions)
    /Users/shuaib/go/src/github.com/IBM/mqcloud-go-sdk/mqcloudv1/mqcloud_v1_integration_test.go:931
------------------------------
•••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••
Ran 240 of 240 Specs in 52.955 seconds
SUCCESS! -- 240 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS
ok      github.com/IBM/mqcloud-go-sdk/mqcloudv1 53.979s
```